{"Functions": {"InterruptedException()": {"Returns": [], "Parameters": [], "Throws": [], "SeeAlso": [], "Permissions": [], "Description": "Constructs an InterruptedException with no detail message.", "history": "Added in API level 1", "FullName": "public InterruptedException ()"}, "InterruptedException(java.lang.String)": {"Returns": [], "Parameters": [["@B_java#lang#String_E@", "String: the detail message."]], "Throws": [], "SeeAlso": [], "Permissions": [], "Description": "Constructs an InterruptedException with the specified detail message.", "history": "Added in API level 1", "FullName": "public InterruptedException (String s)"}}, "Inheritance": ["@B_java#lang#Throwable_E@", "@B_java#lang#Exception_E@"], "ClassName": "java.lang.InterruptedException", "ClassDesc": "Thrown when a thread is waiting, sleeping, or otherwise occupied, and the thread is interrupted, either before or during the activity. Occasionally a method may wish to test whether the current thread has been interrupted, and if so, to immediately throw this exception. The following code can be used to achieve this effect:"}